enhancement(aws_s3 source): Logs processed S3 objects (#22083)
vectordotdev/vector: changelog.d/21128_log_aws_s3_objects_consumed.enhancement.md

> `aws_s3` source now logs s3 objects fetched (and completed if acks are enabled on the sink)

Reviewer: ```suggestion
`aws_s3` source now logs when S3 objects are fetched and when ACKs are enabled, also logs on delivery.
```

Author: I think the suggestion is still ambiguous on what is being logged. I submitted an update with:

> `aws_s3` source now logs when S3 objects are fetched. If ACKs are enabled, it also logs on delivery.
